org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java

```diff
@@ -248,4 +248,10 @@ public final String undoLabel()
 
     return res;
   }
+  public void resetBuffer() {
+    theActions.clear();
+    presentAction=-1;

```

Comment:
@ayeshare - I'm nervous that "-1" is a magic number [1]

I've just looked at the code.  `presentAction` doesn't get initialised. I _think_ our code logic means that would not be an issue, but I'd be happier if it did get assigned something.

So, to solve both of the above I suspect we should introduce a private static member field title `UNSET`, which takes a value of -1.

We can then initialise `presentAction` using it, and we can also set the value using it `resetBuffer`



[1]. https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants

Reply:
this is done now.